### PR TITLE
Update website to 2.1

### PR DESCRIPTION
**Pull request by kwalrath**

### CODE REVIEW

**Comment by kwalrath** [comment]
I still have to make the changes for the mixin example. I plan to add a void discussion soon. (Thought I had a bug for that, but I can't find it.)

**Comment by chalin** [comment]
LG so far.

**Comment by kwalrath** [comment]
Mixin changes are up. I still need to update mentions of Future/Stream and dart:async.

**Comment by kwalrath** [comment]
Staged:
https://kw-www-dartlang-1.firebaseapp.com/guides/language/language-tour#numbers
https://kw-www-dartlang-1.firebaseapp.com/guides/language/language-tour#adding-features-to-a-class-mixins

I'm not confident of my mixin changes. The code works, but I'm not sure I got the text right. @munificent maybe you could take a look?

File: examples/misc/lib/language_tour/classes/orchestra.dart [hunk 1]
```diff
@@ -38,6 +38,7 @@ abstract class Performer {
 // #docregion Musician-and-Maestro
 class Musician extends Performer with Musical {
   // #enddocregion Musician-and-Maestro
+//  Musician() : super('Anonymous');
```
Comment (chalin) [inline examples/misc/lib/language_tour/classes/orchestra.dart]: Was the intent to leave this comment here?

File: examples/misc/lib/language_tour/classes/orchestra.dart [hunk 2]
```diff
@@ -51,11 +52,33 @@ class Maestro extends Person
...
 
+// #docregion mixin-on
+mixin MusicalPerformer on Musician {
+  // #enddocregion mixin-on
+  bool canDance = true;
+
+  @override
+  void entertainMe() {
+    canDance ? dance() : super.entertainMe();
+  }
```
Comment (chalin) [inline examples/misc/lib/language_tour/classes/orchestra.dart]: Make this an arrow definition?

File: src/_guides/language/language-tour.md [hunk 3]
```diff
@@ -3252,15 +3255,26 @@ abstract class Musical {
 }
 {% endprettify %}
 
-<div class="alert alert-info" markdown="1">
-  **Note:** Some restrictions on mixins are being removed. For details, see the
-  [proposed mixin specification.][Dart 2.0 Mixins]
+To specify that only certain types can use the mixin — for example,
+so your mixin can invoke a method that it doesn't define —
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: Use `&mdash;` instead?

File: src/_guides/language/language-tour.md [hunk 4]
```diff
@@ -3252,15 +3255,26 @@ abstract class Musical {
...
+<?code-excerpt "misc/lib/language_tour/classes/orchestra.dart (mixin-on)"?>
+{% prettify dart %}
+mixin MusicalPerformer on Musician {
+  // ···
+}
+{% endprettify %}
+
+<aside class="alert alert-info" markdown="1">
+  **Version note:** Support for the `mixin` keyword was introduced in Dart 2.1.
+  Code in earlier releases usually uses `abstract class` instead.
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: uses -> used

File: src/_guides/language/language-tour.md [hunk 5]
```diff
@@ -3252,15 +3255,26 @@ abstract class Musical {
...
+  For more information on 2.1 mixin changes, see the
+  [Dart SDK changelog][] and [2.1 mixin specification.]
```
Comment (chalin) [inline src/_guides/language/language-tour.md]: [2.1 mixin specification.] -> [2.1 mixin specification.][]

File: examples/misc/lib/language_tour/classes/orchestra.dart [hunk 6]
```diff
@@ -51,11 +52,33 @@ class Maestro extends Person
...
 void main() {
   var director = Maestro('Allen');
   director.entertainMe(); // Waving hands
 
   var musician = Musician('Kathy');
   musician.canPlayPiano = true;
   musician.entertainMe(); // Playing piano
+
+  var singerDancer = SingerDancer('Todd');
+  singerDancer.entertainMe(); // Dancing
```
Comment (chalin) [inline examples/misc/lib/language_tour/classes/orchestra.dart]: You'll probably need to modify `examples/misc/test/language_tour/classes_test.dart`.

Comment (chalin) [reply examples/misc/lib/language_tour/classes/orchestra.dart]: (And you'll need to `allow_failures` for `dev`)

File: examples/misc/lib/language_tour/classes/orchestra.dart (hunk 1 above)
Comment (kwalrath) [reply examples/misc/lib/language_tour/classes/orchestra.dart]: Ha, no. :)

File: src/_guides/language/language-tour.md (hunk 3 above)
Comment (kwalrath) [reply src/_guides/language/language-tour.md]: I'm not crazy about `&mdash;`... it's not very readable.

File: examples/misc/lib/language_tour/classes/orchestra.dart (hunk 6 above)
Comment (kwalrath) [reply examples/misc/lib/language_tour/classes/orchestra.dart]: It's failing for stable/raw too, and I can't figure out why.

File: src/_guides/libraries/library-tour.md [hunk 7]
```diff
@@ -1017,9 +1017,14 @@ use it, import dart:async:
...
 {% prettify dart %}
-// [PENDING: fix the page that uses this!!]
+import 'dart:async';
 {% endprettify %}
 
+<aside class="alert alert-info" markdown="1">
+  **Version note:**
+  As of Dart 2.1, you don't need to import dart:async to
+  use the Future and Stream APIs, because dart:core exports those classes.
+</aside>
```
Comment (chalin) [inline src/_guides/libraries/library-tour.md]: 👍

**Comment by kwalrath** [comment]
Thanks, @chalin!

**Pull request merged**